src/assembler.rs

```diff
@@ -122,7 +122,11 @@ fn make_instruction_map(sbpf_version: SBPFVersion) -> HashMap<String, (Instructi
 
         // Miscellaneous.
         entry("ja", JumpUnconditional, ebpf::JA);
-        entry("syscall", Syscall, ebpf::CALL_IMM);
+        if sbpf_version == SBPFVersion::V1 {
+            entry("syscall", Syscall, ebpf::CALL_IMM);
```

Comment:
The `if` part could be only around the third parameter.

Reply:
Fixed.